sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -657,6 +657,8 @@ primaryExpression
     | base=primaryExpression '.' fieldName=identifier                                          #dereference
     | '(' expression ')'                                                                       #parenthesizedExpression
     | EXTRACT '(' field=identifier FROM source=valueExpression ')'                             #extract
+    | (SUBSTR | SUBSTRING) '(' field=valueExpression (FROM | ',') pos=valueExpression
```

Comment:
The alias `field` seems a bit weird. This can be any string right?

Reply:
How about strExpression?

Reply:
Maybe just `str` or something like that?

Reply:
`str` looks good to me. It is also consistent with the name of class Substring.